File: 6/b.py
import sys

point = [map(int, l.split(", ")) for l in sys.stdin]
minX = min(map(lambda x: x[0], point))
maxX = max(map(lambda x: x[0], point))
minY = min(map(lambda x: x[1], point))
maxY = max(map(lambda x: x[1], point))

c = 0
for x in range(minX, maxX + 1):
  for y in range(minY, maxY + 1):
    dist = sum(map(lambda (px,py): abs(px - x) + abs(py - y), point))
    if dist < 10000:
      c += 1
print c

File: 6/b.clj
(def MAX-DIST 10000)

(defn count-points
  [points]
  (let
    [minX (apply min (map first points))
    maxX (apply max (map first points))
    minY (apply min (map second points))
    maxY (apply max (map second points))]
    (count (filter
      #(< % MAX-DIST)
      (for [x (range minX (+ maxX 1)) y (range minY (+ maxY))]
        (reduce + (map
          (fn [[px py]] (+ (Math/abs (- x px)) (Math/abs (- y py))))
          points)))))))

(let
  [points (map
    (fn [line] (map #(. Integer parseInt %) (clojure.string/split line #", ")))
    (line-seq (java.io.BufferedReader. *in*)))]
  (println (count-points points)))
